salt/pillar/consul_pillar.py

```diff
@@ -267,7 +267,8 @@ def fetch_tree(client, path, expand_keys):
     are folders. Take the remaining data and send it to be formatted
     in such a way as to be used as pillar data.
     '''
-    _, items = consul_fetch(client, path)
+    absolute_path = path.rstrip('/') + '/'
+    _, items = consul_fetch(client, absolute_path)
```

Comment:
Nooo, inside the `consul_fetch`. It should look on its own if the path is right or not. If you reuse it elsewhere, you don't know if trailing slash is needed or not, so you will do it every time you call it, and thus duplicate the code.

Please move that rstrip+/ inside the function. :wink: